libpromises/match_scope.h

```diff
@@ -26,6 +26,10 @@
 #define CFENGINE_MATCH_SCOPE_H
 
 #include <cf3.defs.h>
+#include <regex.h>
```

Comment:
Moving match_scope to libpromises is not complete. Go to `tests/unit/Makefile.am` and update the `cf-agent/match_scope` paths to `libpromises`.

@jimis Do you agree with moving `match_scope` from `cf-agent` to `libpromises` ?